locking/locks.go

```diff
@@ -22,74 +22,88 @@ var (
 	ErrLockAmbiguous = errors.New("lfs: multiple locks found; ambiguous")
 )
 
-// LockFile attempts to lock a file on the given remote name
+// Client is the main interface object for the locking package
+type Client struct {
```

Comment:
This would drastically simplify the calling code for this package (which I think is a 👍 ), but introduces a problem with using the global `config.Config` instance by default, which is a 👎.

I think a compromise could be changing the `getClient` function to take a `*config.Configuration` instance, like so:

```go
var (
        cmu     sync.Mutex
        clients map[*config.Configuration]*client
)

func getClient(cfg *config.Configuration) *client {
        cmu.Lock()
        defer cmu.Unlock()

        if _, ok := clients[cfg]; !ok {
                clients[cfg] = newClient(cfg)
        }

        return clients[cfg]
}
```

and then changing the functions to accept a config argument first, as such:

```go
package locking

func LockFile(cfg *config.Configuration, path string) (id string, err error) {
        return getClient(cfg).LockFile(path)
}
```